tools/SourceKit/cmake/modules/AddSwiftSourceKit.cmake

```diff
@@ -69,7 +69,13 @@ function(add_sourcekit_default_compiler_flags target)
   # TODO(compnerd) this should really use target_compile_options but the use
   # of keyword and non-keyword flags prevents this
   if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
-    list(APPEND c_compile_flags "-fblocks")
+    list(APPEND c_compile_flags -fblocks)
+  elseif(CMAKE_SYSTEM_NAME STREQUAL "Windows")
+    if(SWIFT_COMPILER_IS_MSVC_LIKE)
+      list(APPEND c_compile_flags -Xclang;-fblocks)
```

Comment:
By the way, why the semicolon instead of just a space?

Reply:
Ah, excellent question!  The reason is that `c_compile_flags` is a *list*.  The list will be flattened below.  If we don't append it as a normal list item, it would sometimes get transliterated incorrectly.  Really, if we handled the cross-compilation more to CMake's liking, we could just use `target_compile_options` and avoid a lot of this specialness.

Reply:
I, uh. I know that `c_compile_flags` is a list, but I'm not sure how the following two lines differ in CMake:

```cmake
list(APPEND c_compile_flags -Xclang;-fblocks)
list(APPEND c_compile_flags -Xclang -fblocks)
```

(notice the lack of quotes)

I'm pretty sure we use the latter form elsewhere.